source/docs/essentials/extend-the-graphql-schema.md

```diff
@@ -0,0 +1,281 @@
+</blockquote>
+
+### `contextEnhancer` (optional)
+
+The `contextEnhancer` module definition key should be a function that initializes code that will be made available in GraphQL’s context, under the `loaders` key. **It is specific to Front-Commerce.**
+
+This is where you could construct models or [dataloaders](#TODO) that should be used in your GraphQL resolvers.
+
+The `contextEnhancer` function must return an object whose keys will be merged with previous modules’. It will be passed a single argument with the following keys:
+
+* `req`: the current server request
+* `loaders`: current loaders (from module initialized beforehand)
+* `makeDataLoader`: a factory to build a dataloader (see [dataloaders](#TODO))
+* `config`: the global configuration
+
+Here is an example:
+
+```
+const MessageLoader = require("./loader");
+
+module.exports = {
+  namespace: "Acme/HelloWorld",
+  resolvers: {
+    Query: {
+      sayHello: (_, { name }, { loaders }) => {
+        return loaders.Message.load(`Hello ${name}`);
+      }
+    }
+  }
+  contextEnhancer: ({ req, loaders, makeDataLoader, config }) => {
+    return {
+      Message: MessageLoader(makeDataLoader)(config.apiBaseUrl)
+    }
+  }
+}
+```
+
+## Grouping small modules in a « meta-module »
+
+We recommend to keep your GraphQL modules small and focused on a single feature.
+If several teams are working on the project, small modules are a way to [federate the graph implementation](https://principledgraphql.com/integrity#2-federated-implementation).
+It also makes it easier to change implementations and service providers later in the project.
+For instance you could replace a `Magento2/Cms` module with a `Wordpress/Cms` one without impacting
+the web application (as soon as both schema shares a common API).
+
+But having a lot of different modules could make distribution more tedious.
+That is why Front-Commerce supports the concept of « meta-modules », which are a way to group
+smaller modules together.
+
+For instance, [Front-Commerce’s Magento2 integration](#TODO) can be registered in your project
+using only one line if you want all the features:
+
+```diff
+// .front-commerce.js
+module.exports = {
+  name: "Front-Commerce",
+  url: "https://www.front-commerce.test",
+  modules: [],
+  serverModules: [
+-    { name: "FrontCommerceCore", path: "server/modules/front-commerce-core" }
++    { name: "FrontCommerceCore", path: "server/modules/front-commerce-core" },
++    { name: "Magento2", path: "server/modules/magento2" }
+  ]
+};
+```
+
+Under the hood, it imports several modules at once (`Magento2/Cms`, `Magento2/Catalog`, `Magento2/Checkout`, `Magento2/Search`…).
+
+For a module to be considered as a meta-module, you only need to expose a list of
+sub-modules in the `modules` key of the module definition (along with the namespace).
```

Comment:
We almost had an API like documentation previously but everything wasn't documented (no paragraph about the `modules` keys). I'm wondering if it can be confusing.